python/lsst/ts/eas/diurnal_timer.py

```diff
@@ -0,0 +1,237 @@
+    if now >= local_noon_time:
+        next_day = local_now.date() + datetime.timedelta(days=1)
+        local_noon = datetime.datetime.combine(
+            next_day, datetime.time(12, 0), tzinfo=OBSERVATORY_TIME_ZONE
+        ).astimezone(OBSERVATORY_TIME_ZONE)
+        local_noon_time = Time(local_noon)
+
+    return local_noon_time
+
+
+def get_sun_altitude_deg(t: Time) -> float:
+    """Returns the sun's altitude at the given time.
+
+    The altitude is based on the specified observatory
+    location and does not account for atmospheric
+    refraction.
+
+    Returns
+    -------
+    float
+        The sun's altitude (deg) above the horizon at time t.
+    """
+    sun = get_sun(t)
+    altaz = sun.transform_to(AltAz(obstime=t, location=OBSERVATORY_LOCATION))
+    return altaz.alt.deg
+
+
+def get_crossing_time(target_alt: float) -> Time:
+    """Computes the time when the sun will reach target_alt.
+
+    Parameters
+    ----------
+    target_alt : float
+        The sun altitude at the twilight time to search for, in degrees.
+
+    Returns
+    -------
+    Time
+        The time when the sun will next cross target_alt.
+    """
+
+    def f(t_sec: float) -> float:
+        t = Time(t_sec, format="unix")
+        return get_sun_altitude_deg(t) - target_alt
+
+    # Start by searching for a good window for the root finder.
+    t0 = Time.now() + 1 * u.s
+    sun_altitude_before = get_sun_altitude_deg(t0)
+    for i in range(25):  # Start by searching the next 25 hours.
+        t1 = t0 + 1 * u.hour
+        sun_altitude_after = get_sun_altitude_deg(t1)
+
+        if sun_altitude_before > target_alt and sun_altitude_after < target_alt:
+            break
+
+        # Nope, keep looking
+        sun_altitude_before = sun_altitude_after
+        t0 = t1
+
+    t0 = t0.unix
+    t1 = t1.unix
+    t_cross = brentq(f, t0, t1)
+    return Time(t_cross, format="unix")
+
+
+class DiurnalTimer:
+    """A class representing a timer to wait for noon and for evening twilight.
+
+    Parameters
+    ----------
+    sun_altitude : str | float
+        The sun altitude (degrees) below the horizon that is considered the
+        end of twilight. This can be a number between 0 and -90 or it can
+        be one of "civil" (-6 degrees), "nautical" (-12 degrees), or
+        "astronomical"(-18 degrees).
+    """
+
+    def __init__(self, sun_altitude: str | float = -18.0):
+        if sun_altitude == "civil":
+            sun_altitude = -6.0
+        elif sun_altitude == "nautical":
+            sun_altitude = -12.0
+        elif sun_altitude == "astronomical":
+            sun_altitude = -18.0
+        elif isinstance(sun_altitude, str):
+            raise RuntimeError(
+                "Allowed string values for "
+                "sun_altitude: civil, nautical, astronomical"
+            )
+
+        if not (-90 <= sun_altitude <= 0):
+            raise RuntimeError("sun_altitude not in range -90 to 90")
```

Comment:
This check shall be either -90 to 90, or the errro shall be changed to -90 to 0.